libfreerdp/core/window.c

```diff
@@ -359,33 +359,31 @@ static BOOL update_read_window_state_order(wStream* s, WINDOW_ORDER_INFO* orderI
 
 		Stream_Read_UINT16(s, windowState->numWindowRects); /* numWindowRects (2 bytes) */
 
-		if (windowState->numWindowRects == 0)
+		if (windowState->numWindowRects > 0)
 		{
-			return TRUE;
-		}
+			size = sizeof(RECTANGLE_16) * windowState->numWindowRects;
+			newRect = (RECTANGLE_16*)realloc(windowState->windowRects, size);
 
-		size = sizeof(RECTANGLE_16) * windowState->numWindowRects;
-		newRect = (RECTANGLE_16*)realloc(windowState->windowRects, size);
-
-		if (!newRect)
-		{
-			free(windowState->windowRects);
-			windowState->windowRects = NULL;
-			return FALSE;
-		}
+			if (!newRect)
+			{
+				free(windowState->windowRects);
+				windowState->windowRects = NULL;
+				return FALSE;
+			}
 
-		windowState->windowRects = newRect;
+			windowState->windowRects = newRect;
 
-		if (Stream_GetRemainingLength(s) < 8 * windowState->numWindowRects)
-			return FALSE;
+			if (Stream_GetRemainingLength(s) < 8 * windowState->numWindowRects)
```

Comment:
Sorry, nitpicking on already-merged code, but I wonder whether:

```suggestion
			if (Stream_GetRemainingLength(s) < * windowState->numWindowRects * 8)
```

...could be more consistent with a [similar line](https://github.com/akallabeth/FreeRDP/blob/c4869357dee79cac856738066d2fa1312909d246/libfreerdp/core/window.c#L420) later in this file: 

```c
			if (Stream_GetRemainingLength(s) < windowState->numVisibilityRects * 8)
```

(and potentially safer, in the sense of looking less like a bitshift, and with clearer operator precedence)